### ISSUE TITLE
Recursive structure leads to crash 🙃

### ISSUE THREAD

**Issue by EmCeBeh** [dschick/udkm1Dsim#98]
It is a really stupid thing to do, but when you accidentally add an `AtomMixed` to itself as a structure, the visualiser still works fine, but the (e.g. homogenous) X-ray simulation crashes the entire python kernel without any warning, as something probably gets stuck in a loop.

Not sure if this is possible / worth fixing.

Minimal working example. Line 6 is the problem here, where `SiN` is added as a atom to itself.

```
Si = ud.Atom('Si')
N = ud.Atom('N')
SiN = ud.AtomMixed('SiN')
SiN.add_atom(Si, 0.5)
SiN.add_atom(N,  0.5)
SiN.add_atom(SiN, 1)
layer_SiN    = ud.AmorphousLayer('SiN', 'Layer SiN', 50*u.nm, 3170*u.kg/u.m**3, atom = SiN)
```

**Comment by dschick** [comment]
I guess you tried that _ stupid thing_  just out of curiosity ;)

anyways, good catch and it should be easy to fix when calling `add_atom` in the `AtomMixed` class.

**Comment by dschick** [comment]
solved by PR#133
